### ISSUE TITLE
Updating a liability account fails when passing in opening balance and opening balance date via API

### ISSUE THREAD

**Issue by rene00** [firefly-iii/firefly-iii#5422]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

Hi,

I'm using a go package ([ffgo](https://github.com/rene00/ffgo)) I built from the swagger API to migrate my personal finance data into firefly. When trying to update a liability account with an opening balance and date, I'm receiving a 500 with this error message:
```
 [2021-12-24 10:21:14] local.ERROR: Exception is: {"class":"ErrorException","errorMessage":"Undefined array key \"liability_direction\"","time":"Fri, 24 Dec 2021 10:21:14 +0000","
file":"\/var\/www\/html\/app\/Services\/Internal\/Update\/AccountUpdateService.php","line":328,"code":0,"version":"5.6.8","url":"http:\/\/10.0.1.13:8080\/api\/v1\/accounts\/53841","userAgent"
:null,"json":true}
```
I'm unable to reproduce this error on the demo site.

Looks like the error I'm receiving is triggered [here](https://github.com/firefly-iii/firefly-iii/blob/main/app/Services/Internal/Update/AccountUpdateService.php#L328-L330). From what I can tell from the [schema](https://api-docs.firefly-iii.org/#/accounts/updateAccount) for this endpoint, I'm unable to pass `liability_direction` within the body of the request.

When I don't set `opening_balance` and `opening_date`, the request is successful.

### Expected behaviour

An updated account with opening balance and date set

### Debug information

```
firefly    | [2021-12-24 10:21:14] local.ERROR: Exception is: {"class":"ErrorException","errorMessage":"Undefined array key \"liability_direction\"","time":"Fri, 24 Dec 2021 10:21:14 +0000","file":"\/var\/www\/html\/app\/Services\/Internal\/Update\/AccountUpdateService.php","line":328,"code":0,"version":"5.6.8","url":"http:\/\/127.0.0.1:8080\/api\/v1\/accounts\/53841","userAgent":null,"json":true}  
firefly    | [2021-12-24 10:21:14] local.ERROR: Exception when mailing: Address in mailbox given [] does not comply with RFC 2822, 3.6.2.  
firefly    | [2021-12-24 10:21:14] local.ERROR: Undefined array key "liability_direction" {"userId":1,"exception":"[object] (ErrorException(code: 0): Undefined array key \"liability_direction\" at /var/www/html/app/Services/Internal/Update/AccountUpdateService.php:328)
firefly    | [stacktrace]
firefly    | #0 /var/www/html/app/Services/Internal/Update/AccountUpdateService.php(328): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError()
firefly    | #1 /var/www/html/app/Services/Internal/Update/AccountUpdateService.php(107): FireflyIII\\Services\\Internal\\Update\\AccountUpdateService->updateCreditLiability()
firefly    | #2 /var/www/html/app/Repositories/Account/AccountRepository.php(780): FireflyIII\\Services\\Internal\\Update\\AccountUpdateService->update()
firefly    | #3 /var/www/html/app/Api/V1/Controllers/Models/Account/UpdateController.php(79): FireflyIII\\Repositories\\Account\\AccountRepository->update()
firefly    | #4 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(54): FireflyIII\\Api\\V1\\Controllers\\Models\\Account\\UpdateController->update()
firefly    | #5 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(45): Illuminate\\Routing\\Controller->callAction()
firefly    | #6 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Route.php(262): Illuminate\\Routing\\ControllerDispatcher->dispatch()
firefly    | #7 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Route.php(205): Illuminate\\Routing\\Route->runController()
firefly    | #8 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(695): Illuminate\\Routing\\Route->run()
firefly    | #9 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}()
firefly    | #10 /var/www/html/app/Api/V1/Controllers/Models/Account/UpdateController.php(58): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
firefly    | #11 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): FireflyIII\\Api\\V1\\Controllers\\Models\\Account\\UpdateController->FireflyIII\\Api\\V1\\Controllers\\Models\\Account\\{closure}()
firefly    | #12 /var/www/html/app/Api/V1/Controllers/Controller.php(67): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
firefly    | #13 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): FireflyIII\\Api\\V1\\Controllers\\Controller->FireflyIII\\Api\\V1\\Controllers\\{closure}()
firefly    | #14 /var/www/html/app/Http/Middleware/Binder.php(78): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
firefly    | #15 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): FireflyIII\\Http\\Middleware\\Binder->handle()
firefly    | #16 /var/www/html/app/Http/Middleware/Authenticate.php(75): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
firefly    | #17 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): FireflyIII\\Http\\Middleware\\Authenticate->handle()
firefly    | #18 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
firefly    | #19 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(697): Illuminate\\Pipeline\\Pipeline->then()
firefly    | #20 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(672): Illuminate\\Routing\\Router->runRouteWithinStack()
firefly    | #21 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(636): Illuminate\\Routing\\Router->runRoute()
firefly    | #22 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(625): Illuminate\\Routing\\Router->dispatchToRoute()
firefly    | #23 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(167): Illuminate\\Routing\\Router->dispatch()
firefly    | #24 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}()
firefly    | #25 /var/www/html/app/Http/Middleware/InstallationId.php(52): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
firefly    | #26 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): FireflyIII\\Http\\Middleware\\InstallationId->handle()
firefly    | #27 /var/www/html/vendor/fideloper/proxy/src/TrustProxies.php(57): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
firefly    | #28 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fideloper\\Proxy\\TrustProxies->handle()
firefly    | #29 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
firefly    | #30 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php(31): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
firefly    | #31 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsToNull->handle()
firefly    | #32 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
firefly    | #33 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php(40): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
firefly    | #34 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\TrimStrings->handle()
firefly    | #35 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
firefly    | #36 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle()
firefly    | #37 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(86): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
firefly    | #38 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance->handle()
firefly    | #39 /var/www/html/app/Http/Middleware/SecureHeaders.php(51): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
firefly    | #40 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): FireflyIII\\Http\\Middleware\\SecureHeaders->handle()
firefly    | #41 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
firefly    | #42 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(142): Illuminate\\Pipeline\\Pipeline->then()
firefly    | #43 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(111): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter()
firefly    | #44 /var/www/html/public/index.php(76): Illuminate\\Foundation\\Http\\Kernel->handle()
firefly    | #45 {main}
```

**Comment by JC5** [comment]
Nice find, I'll pick it up :+1:

**Comment by 3ssen** [comment]
I'll have the same sort of bug after migrating to 5.6.9.

When I'm in the edit liability screen I cannot change the name anymore unless I set the startdate of the startbalance. The balance is set to 0. After saving the startdate is back to 00-00-0000, probably not saved because the startbalance is 0.

On the demo site the startbalance field is empty instead of 0 and then there isn't a validation error on the startbalance.

Hope this helps fixing the bug.

**Comment by JC5** [comment]
That last part is fixed in 5.6.10

**Comment by JC5** [comment]
Issue will be fixed in the next release, thanks for reporting :+1:
